### PR TITLE
refactor: use es6 classes

### PR DESCRIPTION
**Pull request by kontrollanten**
#92 but with the conflicts solved:

> Adds babel and uses es6 classes instead of "class.extend".

### CODE REVIEW

**Comment by kontrollanten** [comment]
@yokuze @jthomerson Can I do something to help make this PR merged? We've some other features depending on this, which will add support for PeerTube's HLS to get working with videojs-chromecast.

**Review by yokuze — CHANGES_REQUESTED** [review]
This looks good, thank you for the contribution @kontrollanten. Sorry it sat for so long. Just a couple of things before we can merge:

1. Could you rebase this and fix any merge conflicts?
2. Could you add a `.browserslistrc` file with the contents:

```
defaults
> 0.05%
not dead
not safari < 9
not ios < 9
not op_mini all
not android < 4.4
```

And the one review comment.

File: package.json [hunk 1]
```diff
@@ -27,14 +27,21 @@
   },
   "homepage": "https://github.com/silvermine/videojs-chromecast#readme",
   "dependencies": {
-    "class.extend": "0.9.1",
     "webcomponents.js": "git+https://git@github.com/webcomponents/webcomponentsjs.git#v0.7.24"
   },
   "devDependencies": {
+    "@babel/core": "^7.13.16",
```
Comment (yokuze) [inline package.json:33]: @kontrollanten could you pin these to exact versions using the `--save-exact` flag? Thanks

**Review by yokuze — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (kontrollanten) [reply package.json]: By using package-lock.json file there should not be any reason to pin the dependencies in the package.json file. But I pinned them anyway just to be consistent. :)

**Comment by kontrollanten** [comment]
Thanks for the review, now everything is rebase and should be fixed.

**Review by yokuze — APPROVED** [review]

**Comment by yokuze** [comment]
Thanks @kontrollanten, and thanks @izkmdz for your testing and review.

**Comment by dwassing** [comment]
@yokuze Will you cut a new npm release with this change?

**Pull request merged**